services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/gdpr/KortistoGdprServiceITest.kt

```diff
@@ -319,12 +335,245 @@ class KortistoGdprServiceITest(
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.HAKEMUSASIOINTI)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(KAYTTAJA_INPUT_HAKIJA.sukunimi)
+            }
+        }
+
+        @Test
+        fun `throws and exception when the user is the only admin and there's an active application without the user`() {
+            val hanke =
+                hankeFactory.builder(USERID).withHankealue().saveWithYhteystiedot {
+                    val kayttaja =
+                        hankeKayttajaService.getKayttajaByUserId(this.hankeEntity.id, USERID)!!
+                    omistaja { addYhteyshenkilo(it, kayttaja) }
+                }
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).inHandling().saveWithYhteystiedot {
+                    hakija(kayttooikeustaso = Kayttooikeustaso.HAKEMUSASIOINTI)
+                }
+
             val failure = assertFailure { gdprService.deleteInfo(USERID) }
 
-            failure.hasClass(NotImplementedError::class)
+            failure
+                .isInstanceOf(DeleteForbiddenException::class)
+                .prop(DeleteForbiddenException::errors)
+                .single()
+                .prop(GdprError::message)
+                .prop(LocalizedMessage::fi)
+                .contains(hakemus.applicationIdentifier!!)
+        }
+
+        @Test
+        fun `deletes hankekayttaja when there's another admin in the hanke`() {
+            lateinit var kayttaja: HankekayttajaEntity
+            val hanke =
+                hankeFactory.builder(USERID).withHankealue().saveWithYhteystiedot {
+                    kayttaja =
+                        hankeKayttajaService.getKayttajaByUserId(this.hankeEntity.id, USERID)!!
+                    omistaja { addYhteyshenkilo(it, kayttaja) }
+                }
+            val hakemus =
+                hakemusFactory.builder(USERID, hanke).inHandling().saveWithYhteystiedot {
+                    hakija(kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET)
+                }
+
+            gdprService.deleteInfo(USERID)
+
+            assertThat(hankeRepository.findAll()).single().prop(HankeEntity::id).isEqualTo(hanke.id)
+            assertThat(applicationRepository.findAll())
+                .single()
+                .prop(ApplicationEntity::id)
+                .isEqualTo(hakemus.id)
+            assertThat(hankeKayttajaService.getKayttajatByHankeId(hanke.id)).single().all {
+                prop(HankeKayttajaDto::kayttooikeustaso).isEqualTo(Kayttooikeustaso.KAIKKI_OIKEUDET)
+                prop(HankeKayttajaDto::sukunimi).isEqualTo(KAYTTAJA_INPUT_HAKIJA.sukunimi)
```

Comment:
Perhaps it would be more clear where this `KAYTTAJA_INPUT_HAKIJA` comes from if above `hakemus` was created like this:
```
val hakemus = 
    hakemusFactory.builder(USERID, hanke).inHandling().saveWithYhteystiedot {
        hakija(
            hakija = KAYTTAJA_INPUT_HAKIJA,
            kayttooikeustaso = Kayttooikeustaso.KAIKKI_OIKEUDET
        )
```

Reply:
Thank you for the suggestion, I added the explicit kayttaja input parameter.